music-player/src/reducers.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  ADD_PLAYLIST_ACTION_TYPE,
+  ADD_SONG_TO_PLAYLIST_ACTION_TYPE,
+  CHANGE_PLAYLIST_ACTION_TYPE,
+  PLAY_NEXT_SONG_ACTION_TYPE,
+  PLAY_PREV_SONG_ACTION_TYPE,
+  REMOVE_SONG_FROM_PLAYLIST_ACTION_TYPE,
+} from './actions';
+import { songsNamesList } from './data';
+import { Action, SongsListState, State } from './types';
+import { routerReducer } from 'react-router-redux';
+import { combineReducers } from 'redux';
+import { viewport } from './forUrl/viewport';
+
+const initialSongsListState: SongsListState = {
+  currentSongIndex: 0,
+  currentPlaylistName: 'All',
+  playlists: {
+    All: songsNamesList.map((name, index) => index),
+    FirstSongs: [0, 1],
+  },
+  isValidURL: true
+};
+
+const songsListStateReducer = (songsListState: SongsListState = initialSongsListState, action: Action) => {
```

Comment:
This could definitely benefit from splitting into smaller reducers.

Consider the following state:
```
{
playlists: playlist[]
songByID: {[id] : songInformation}
}
```

Where a playlist would have:
Playlist: {songs: songId[], name, etc..}

Reply:
I'm still having problems with dividing it into two reducers:
`{
current: {  
    currentSongIndex: number;
    currentPlaylistName: string;
    isValidUrl: boolean;
},
playlists: {
    playlists: {[s: string]: number[]};
}}`

For example: With the action 'CHANGE_PLAYLIST', we change `currentSongIndex` to the first song of the new playlist. In order to do that we must have access to both `playlists` and `current`.
We can do this in the component by sending all the playlists to the component but letting the components have all the playlists is a bit much. So it sounds like it should be in the reducer, isn't it?